api/src/main/java/org/pmiops/workbench/api/UserMetricsController.java

```diff
@@ -97,7 +97,16 @@ public void setDistinctWorkspaceLimit(int limit) {
   public ResponseEntity<RecentResource> updateRecentResource(String workspaceNamespace, String workspaceId, RecentResourceRequest recentResourceRequest) {
     Timestamp now = new Timestamp(clock.instant().toEpochMilli());
     long wId = getWorkspaceId(workspaceNamespace, workspaceId);
-    UserRecentResource recentResource = userRecentResourceService.updateNotebookEntry(wId, userProvider.get().getUserId(), recentResourceRequest.getNotebookName(), now);
+    String notebookPath;
+    if (recentResourceRequest.getNotebookName().startsWith("gs://")) {
```

Comment:
Does this need a null check? Could this be called with a recent resource that is not a notebook?

Reply:
nope. No null check needed.